cli/utilities/packages.py

```diff
@@ -0,0 +1,181 @@
+
+        Args:
+            username (str): username to use when authorizing against the server
+            password (str): password to use when authorizing against the server
+            serverurl (str): server URL
+            baseurl (str): base URL for content
+            force (bool): register the system even if it is already registered
+        """
+        cmd = f"{self.base_cmd} register --username {username} --password {password}"
+        if serverurl:
+            cmd += f" --serverurl {serverurl}"
+
+        if baseurl:
+            cmd += f" --baseurl {baseurl}"
+
+        if force:
+            cmd += " --force"
+
+        return self.execute(sudo=True, long_running=True, cmd=cmd)
+
+    def unregister(self):
+        """Unregister this system from the Customer Portal or another subscription management service."""
+        cmd = f"{self.base_cmd} unregister"
+
+        return self.execute(sudo=True, long_running=True, cmd=cmd)
+
+
+class Repos(Cli):
+    """This module provides CLI interface to perform subscription manager repo operations."""
+
+    def __init__(self, nodes, base_cmd):
+        super(Repos, self).__init__(nodes)
+
+        self.base_cmd = f"{base_cmd} repos"
+
+    def enable(self, repos):
+        """Enable subscription manager repos.
+
+        Args:
+            repos (str|list): repositories to be enabled
+        """
+        if isinstance(repos, str):
+            repos = [str(repos)]
+
+        repos = list(map(lambda repo: f"--enable={repo}", repos))
+        cmd = f"{self.base_cmd} {' '.join(repos)}"
+
+        return self.execute(sudo=True, long_running=True, cmd=cmd)
+
+    def disable(self, repos):
+        """Enable subscription manager repos.
+
+        Args:
+            repos (str|list): repositories to be disabled
+        """
+        if isinstance(repos, str):
+            repos = [str(repos)]
+
+        repos = list(map(lambda repo: f"--disable={repo}", repos))
+        cmd = f"{self.base_cmd} {' '.join(repos)}"
+
+        return self.execute(sudo=True, long_running=True, cmd=cmd)
+
+    def list(self, status=None):
+        """List all/enabled/disabled repositories
+
+        Args:
+            status (str): repositories status (all/enabled/disabled)
+        """
+        cmd = f"{self.base_cmd} --list"
+        if status:
+            cmd += f"-{status}"
+
+        out = self.execute(sudo=True, cmd=cmd)
+        if isinstance(out, tuple):
+            return out[0].strip()
+        return out
+
+
+class Rpm(Cli):
+    """This module provides CLI interface for rpm operations"""
+
+    def __init__(self, nodes):
+        super(Rpm, self).__init__(nodes)
+        self.base_cmd = "rpm"
+
+    def query(self, pkg):
+        """query a package information
+
+        Args:
+            pkg (str): package name
+        """
+        cmd = f"{self.base_cmd} --query {pkg}"
+
+        return self.execute(sudo=True, long_running=True, cmd=cmd)
```

Comment:
My two cents on this module... you have rightly classified it as a utility hence, handling the unexpected behavior needs to occur within this module. On event of such behavior an exception must be raised.

Reply:
done